kernel/thread.c

```diff
@@ -474,8 +474,8 @@ static char *setup_thread_stack(struct k_thread *new_thread,
 
 	LOG_DBG("stack %p for thread %p: obj_size=%zu buf_start=%p "
 		" buf_size %zu stack_ptr=%p",
-		stack, new_thread, stack_obj_size, stack_buf_start,
-		stack_buf_size, stack_ptr);
+		stack, new_thread, stack_obj_size, (void *)stack_buf_start,
+		stack_buf_size, (void *)stack_ptr);
```

Comment:
What's the error being suppressed here?  In general all pointers will promote to void* automatically[1], right?  Can't the function call on the other side of LOG_DBG() just have its signature changed?

[1] Cue someone coming in to explain how that's not true in C++ or C11 or whatever.

Reply:
It is only for logging. 
See cbrpintf docs:
```
It is recommended to cast any character pointer used with %p format specifier to other pointer type (e.g. void *). If format string is not accessible then only static packaging is possible and it will append all detected strings. Character pointer used for %p will be considered as string pointer. Copying from unexpected location can have serious consequences (e.g., memory fault or security violation).
```
So currently logging prints warning if `char *` is found to be used with `%p` so it can be casted.
When i enabled debug level in that module I got that warning here.